e2e/client/types.go

```diff
@@ -0,0 +1,295 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	netclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
+
+	"github.com/maiqueb/multus-dynamic-networks-controller/pkg/annotations"
+)
+
+type E2EClient struct {
+	k8sClient          kubernetes.Interface
+	netAttachDefClient netclient.K8sCniCncfIoV1Interface
+}
+
+func New(config *rest.Config) (*E2EClient, error) {
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	netClient, err := netclient.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &E2EClient{
+		k8sClient:          clientSet,
+		netAttachDefClient: netClient,
+	}, nil
+}
+
+func (c *E2EClient) AddNetAttachDef(netattach *nettypes.NetworkAttachmentDefinition) (*nettypes.NetworkAttachmentDefinition, error) {
+	return c.netAttachDefClient.NetworkAttachmentDefinitions(netattach.ObjectMeta.Namespace).Create(context.TODO(), netattach, metav1.CreateOptions{})
+}
+
+func (c *E2EClient) DelNetAttachDef(namespace string, podName string) error {
+	return c.netAttachDefClient.NetworkAttachmentDefinitions(namespace).Delete(context.TODO(), podName, metav1.DeleteOptions{})
+}
+
+func (c *E2EClient) AddNamespace(name string) (*corev1.Namespace, error) {
+	return c.k8sClient.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}, metav1.CreateOptions{})
+}
+
+func (c *E2EClient) DeleteNamespace(name string) error {
+	const timeout = 30 * time.Second
+
+	if err := c.k8sClient.CoreV1().Namespaces().Delete(context.TODO(), name, metav1.DeleteOptions{}); err != nil {
```

Comment:
I'm pretty sure there might be one - but since we re-use the same static namespace, it will not be wise to proceed the tests until the namespace is cleared. 

Hence, I'd rather leave this as it is.

I'll eventually want to change the test framework to use ephemeral non-reusable namespaces, but, now's not the time for that.